disorder-fsm/src/Disorder/FSM.hs

```diff
@@ -1,9 +1,14 @@
 {-# LANGUAGE NoImplicitPrelude #-}
 {-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
```

Comment:
I am quick sure that for [such a trivial case](https://github.com/ambiata/disorder.hs/blob/topic/cont-fsm/disorder-fsm/src/Disorder/FSM.hs#L54-L63) it would generate correct instances from the day it was added to GHC. And [similar code](https://github.com/ambiata/amazonka/blob/ambiata/amazonka/src/Control/Monad/Trans/AWS.hs#L153-L164) is already in production.